grades/models.py

```diff
@@ -24,7 +24,7 @@ class Grade(models.Model):
 
     @classmethod
     def addGradeUserCourse(cls, user, courseID, grade, semester, year):
-        newGrade = Grade.object.create()
+        newGrade = Grade.objects.create()
```

Comment:
Another thought is you might want to use newgrade = Grade() and not Grade.objects.create(). If this works (I'm not sure it will), it will actually create a record with no fields other than the primary key, then when you add the additional fields and call save(), that will then do an update, so it's 2 DB calls instead of 1.

Reply:
In the Grade model you are allowing both User and Course to be null and blank. Can you think of a situation where you would want to have a grade record but not know who the user is or what course it is?

Reply:
@simzacks I have changed the models as you mentioned above.
About your last point I did it as it has shown in your presentation about the AUTH_USER_MODEL field.